online-shop/src/app/all-products/all-products.component.html

```diff
@@ -0,0 +1,10 @@
+<p>all-products works!</p>
+
+<router-outlet></router-outlet>
```

Comment:
you shouldn't be needing <router-outlet> here. the one in the app.component.html suffices. 
it usually makes sense to have multiple components with router-outlet when they'd need common parent components that differ a lot from one another, for example:
- all the login, sign up, forgot password, etc to have one parent with a pretty empty html, perhaps only containing the router-outlet
- all pages after the user is logged in to have another common parent, with header, menu, content and footer containing the header, menu, router-outlet for content, and the footer.